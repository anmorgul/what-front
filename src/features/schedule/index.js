export { AllSchedules } from './shedule-all/index.js';
export { ScheduleGroup } from './schedule-group/index.js';
export { AddSchedule } from './add-schedule/index.js';
