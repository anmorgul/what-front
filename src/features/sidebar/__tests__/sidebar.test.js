import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import { BrowserRouter as Router, useHistory } from 'react-router-dom';
import { Sidebar } from '@/features';
import { useSelector } from 'react-redux';
import { paths } from '@/shared';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn().mockReturnValue({
    push: jest.fn(),
    location: { pathname: '/lessons' }
  }),
}));

describe('Sidebar', () => {
  it('', () => {
    useSelector.mockReturnValue({ currentUser: { role: 2 } });
    const { container } = render(
      <Router>
        <Sidebar />
      </Router>
    );
    const sidebarContainer = container.getElementsByClassName('sidebar');
    expect(sidebarContainer).toMatchSnapshot();
  });

  it('should navigate to lessons page', async () => {
    const history = useHistory();
    useSelector.mockReturnValue({ currentUser: { role: 2 } });
    const { getByText } = render(
      <Router>
        <Sidebar />
      </Router>
    );

    const navigate_link = getByText(/Lessons/i);

    await waitFor(() => {
      fireEvent.click(navigate_link);
    });
    expect(history.location.pathname).toBe(paths.LESSONS);
  });

  it('schedule should not be displayed', () => {
    useSelector.mockReturnValue({ currentUser: { role: 2 } });
    const { queryByText } = render(
      <Router>
        <Sidebar />
      </Router>
    );

    const groups_link = queryByText(/Schedule/i);

    expect(groups_link).toBeNull();
  });
});