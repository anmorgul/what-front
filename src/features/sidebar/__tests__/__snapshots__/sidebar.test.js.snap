// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar  1`] = `HTMLCollection []`;
